File: oop/lab3/hell/Circle.java
import java.lang.Math;

public class Circle extends Figure {
    private double radius;

    public Circle(double radius) throws IllegalArgumentException {
        if (radius <= 0) {
            throw new IllegalArgumentException("Radius must be positive.");
        }
        this.radius = radius;
    }

    @Override
    public double area() {
        return Math.PI * radius * radius;
    }

    @Override
    public double perimeter() {
        return 2 * Math.PI * radius;
    }

    @Override
    public String name() {
        return "Circle";
    }
}

File: 2024_semester_2/oop/lab3/cpp/src/circle.cpp
#include <stdexcept>

#include "circle.hpp"

Circle::Circle(double radius) noexcept(false)
{
    if (radius <= 0)
    {
        throw std::invalid_argument("Radius must be greater than 0.");
    }
    this->radius = radius;
}

double Circle::area() const noexcept(true)
{
    return M_PI * this->radius * this->radius;
}

double Circle::perimeter() const noexcept(true)
{
    return 2 * M_PI * this->radius;
}

const char *Circle::name() const noexcept(true)
{
    return "Circle";
}
